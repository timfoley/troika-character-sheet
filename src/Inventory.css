.inventory {
    grid-area: inventory;
}

.inventory__grid {
    display           : grid;
    grid-template-rows: repeat(12, 1fr);
    grid-auto-flow    : column;
    row-gap           : 8px;
    column-gap        : 32px;
    padding-left      : 20px;
}

@media (min-width: 500px) {
    .inventory__grid {
        grid-template-rows   : repeat(6, 1fr);
        grid-template-columns: 1fr 1fr;
    }
}

.item {
    border-bottom: 1px solid black;
}

.item>input {
    height       : 100%;
    width        : 100%;
    padding      : 2px;
    border       : none;
    border-radius: 0;
}