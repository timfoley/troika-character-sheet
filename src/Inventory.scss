.inventory {
	grid-area: inventory;
}

.inventory__grid {
	display: grid;
	grid-template-rows: repeat(12, 36px);
	grid-auto-flow: column;
	row-gap: 4px;
	column-gap: 50px;
}

@media (min-width: 500px) {
	.inventory__grid {
		grid-template-rows: repeat(6, 40px);
		grid-template-columns: 1fr 1fr;
	}
}

.item {
	border-bottom: 1px solid black;
}

.item__input {
	height: 100%;
	width: 100%;
	padding: 2px;
	border: none;
	border-radius: 0;
	resize: none;
	margin-bottom: -10px;
}
