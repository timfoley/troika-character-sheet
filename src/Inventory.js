import React from "react";
import PropTypes from "prop-types";

import "./Inventory.css";

export function Inventory(props) {
	const items = [];

	const updateInventory = (index, value) => {
		const newInventory = props.inventory;
		newInventory[index] = {
			name: value,
			slots: 1,
			notes: "",
		};
		props.updateCharacter("inventory", newInventory);
	};

	for (let i = 0; i < 12; i++) {
		if (props.inventory[i]) {
			items.push(
				<React.Fragment key={"item" + i}>
					<li className="item">
						<input
							id={i}
							type="text"
							aria-label={`Slot ${i + 1}`}
							defaultValue={props.inventory[i].name}
							onChange={e => updateInventory(i, e.target.value)}
						/>
					</li>
				</React.Fragment>
			);
		} else {
			items.push(
				<React.Fragment key={"item" + i}>
					<li className="item blank">
						<input
							id={i}
							type="text"
							aria-label={`Empty slot ${i + 1}`}
							defaultValue=""
							onChange={e => updateInventory(i, e.target.value)}
						/>
					</li>
				</React.Fragment>
			);
		}
	}

	return (
		<div className="inventory">
			<h2>Inventory</h2>
			<ol className="inventory__grid">{items}</ol>
		</div>
	);
}

Inventory.propTypes = {
	inventory: PropTypes.array,
};

Inventory.defaultProps = {
	inventory: [{ name: "", notes: "", quantity: 1, slots: 1 }],
};
