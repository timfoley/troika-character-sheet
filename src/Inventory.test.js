import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";

import { Inventory } from "./Inventory";

it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Inventory />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
	const { container } = render(<Inventory />);
	expect(container.firstChild).toMatchSnapshot();
});
