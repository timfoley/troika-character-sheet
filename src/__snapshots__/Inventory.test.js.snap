// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="inventory"
>
  <h2>
    Inventory
  </h2>
  <ol
    class="inventory__grid"
  >
    <li
      class="item"
    >
      <input
        aria-label="Slot 1"
        id="0"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 2"
        id="1"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 3"
        id="2"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 4"
        id="3"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 5"
        id="4"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 6"
        id="5"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 7"
        id="6"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 8"
        id="7"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 9"
        id="8"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 10"
        id="9"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 11"
        id="10"
        type="text"
        value=""
      />
    </li>
    <li
      class="item blank"
    >
      <input
        aria-label="Empty slot 12"
        id="11"
        type="text"
        value=""
      />
    </li>
  </ol>
</div>
`;
